torch/csrc/distributed/rpc/init.cpp

```diff
@@ -331,23 +344,68 @@ PyObject* rpc_init(PyObject* /* unused */) {
   // pythonRpcHandler is cleaned up in shutdown(), after
   // shutdown(), python objects returned from rpc python call can not be
   // resolved.
-  auto future = shared_ptr_class_<FutureIValue>(module, "Future")
-                    .def(
-                        "wait",
-                        [&](FutureIValue& fut) {
-                          auto& pythonRpcHandler =
-                              PythonRpcHandler::getInstance();
-                          const auto& value = fut.wait();
-                          pybind11::gil_scoped_acquire ag;
-                          auto obj = torch::jit::toPyObject(value);
-                          pythonRpcHandler.handleException(obj);
-                          return obj;
-                        },
-                        py::call_guard<py::gil_scoped_release>(),
-                        R"(
+  shared_ptr_class_<FutureIValue>(module, "Future")
+      .def(
+          "wait",
+          [&](FutureIValue& fut) {
+            auto& pythonRpcHandler = PythonRpcHandler::getInstance();
+            const auto& value = fut.wait();
+            pybind11::gil_scoped_acquire ag;
+            auto obj = torch::jit::toPyObject(value);
+            pythonRpcHandler.handleException(obj);
+            return obj;
+          },
+          py::call_guard<py::gil_scoped_release>(),
+          R"(
 Wait on future to complete and return the object it completed with.
 If the future completes with an error, an exception is thrown.
-              )");
+          )")
+      .def(
+          "add_done_callback",
+          [&](FutureIValue& fut, py::function cb) {
```

Comment:
Should we have some validation here to ensure that cb is valid? For example its not None and it has the correct signature?